modules/registry-refactoring/src/main/java/org/apache/airavata/registry/core/repositories/appcatalog/StorageResourceRepository.java

```diff
@@ -0,0 +1,197 @@
+        }
+    }
+
+    @Override
+    public List<StorageResourceDescription> getStorageResourceList(Map<String, String> filters) throws AppCatalogException {
+        try {
+            if (filters.containsKey(DBConstants.StorageResource.HOST_NAME)) {
+                Map<String,Object> queryParameters = new HashMap<>();
+                queryParameters.put(DBConstants.ComputeResource.HOST_NAME, filters.get(DBConstants.StorageResource.HOST_NAME));
+                List<StorageResourceDescription> storageResourceDescriptionList = select(QueryConstants.FIND_STORAGE_RESOURCE, -1, 0, queryParameters);
+
+                return storageResourceDescriptionList;
+            }
+            else {
+                logger.error("Unsupported field name for compute resource.", new IllegalArgumentException());
+                throw new IllegalArgumentException("Unsupported field name for compute resource.");
+            }
+        } catch (Exception e) {
+            logger.error("Error while retrieving storage resource list", e);
+            throw new AppCatalogException(e);
+        }
+    }
+
+    @Override
+    public List<StorageResourceDescription> getAllStorageResourceList() throws AppCatalogException {
+        try {
+            return select(QueryConstants.FIND_ALL_STORAGE_RESOURCES, 0);
+        } catch (Exception e) {
+            logger.error("Error while retrieving storage resource list", e);
+            throw new AppCatalogException(e);
+        }
+    }
+
+    @Override
+    public Map<String, String> getAllStorageResourceIdList() throws AppCatalogException {
+        try {
+            Map<String, String> storageResourceMap = new HashMap<String, String>();
+            List<StorageResourceDescription> storageResourceDescriptionList = select(QueryConstants.FIND_ALL_STORAGE_RESOURCES, 0);
+            if (storageResourceDescriptionList != null) {
+                for (StorageResourceDescription storageResourceDescription: storageResourceDescriptionList) {
+                    storageResourceMap.put(storageResourceDescription.getStorageResourceId(), storageResourceDescription.getHostName());
+                }
+            }
+            return storageResourceMap;
+        } catch (Exception e) {
+            logger.error("Error while retrieving storage resource ID list", e);
+            throw new AppCatalogException(e);
+        }
+    }
+
+    @Override
+    public Map<String, String> getAvailableStorageResourceIdList() throws AppCatalogException {
+        try {
+            Map<String, String> storageResourceMap = new HashMap<String, String>();
+            List<StorageResourceDescription> storageResourceDescriptionList = select(QueryConstants.FIND_ALL_STORAGE_RESOURCES, 0);
+            if (storageResourceDescriptionList != null) {
+                for (StorageResourceDescription storageResourceDescription: storageResourceDescriptionList) {
+                    if (storageResourceDescription.isEnabled()) {
```

Comment:
It would be great to filter this out in the query itself. Then the code here is the same as on 130 and can be factored out.

Reply:
+1